### DISCUSSION TITLE
How to get the bounding rect of a Vec of shapes?

### DISCUSSION THREAD

**Discussion by nk9** [georust/geo#1179] [Q&A] [answered]
I have a Vec of MultiPolygons.  What is the best way to get a bounding rect which contains them all?

I thought it would be as simple as using `union` on all of them, but `bounding_rect` returns a `Rect` whereas `union` returns a `Polygon`. (And possibly a `MultiPolygon` when used on one of those?)

**Comment by mthh** [comment, marked as answer]
I think you can use a [`GeometryCollection`](https://docs.rs/geo-types/latest/geo_types/geometry/struct.GeometryCollection.html) (which is a collection of Geometries) for this.

```rust
let vec_mpl: Vec<MultiPolygon<f32>> = vec![mpl1, mpl2];
let gc = GeometryCollection::new_from(vec_mpl.into_iter().map(Geometry::from).collect());
println!("{:?}", gc.bounding_rect());
```
